### PR TITLE
Add GUI helper to hist for minimum and maximum. Fixes #3880

### LINKED ISSUE #3880: Add "gears" to hist GUI to see what the computed min/max values are
**Description**  

It would be helpful to see what the computed min and max values are from within the _hist_ GUI.  These values would help one pick a reasonable value for NBINS and also give an indication that there are some crazy outliers hiding in the data ready to mess things up.  
**Example**  

I'm thinking of the little clickable symbol that looks like a gear that we have in applications like _maptemplate_ or _cam2map_.

### PR DESCRIPTION
**Pull request by scsides**
…N values are of the specified band of the cube

Add a GUI helper to hist that calculates the minimum and maximum DN of the first band of a cube (Cube Attributes are used), and populates the parameters with the respective values.

## Related Issue
#3880

## Motivation and Context

Help the user find a good min/max without having to run stats on the cube before running hist

## How Has This Been Tested?

This is a GUI feature. Runs were made with several cubes, and the GUI was changed to the expected state.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [ X I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [ ] I have added myself to the [AUTHORS.rst](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/AUTHORS.rst) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by krlberry — APPROVED** [review]
:+1: Looks good to me!

**Comment by krlberry** [comment]
Actually, could you add a note about the change to the new changelog? I forgot about this earlier.

File: isis/src/base/apps/hist/main.cpp [hunk 1]
```diff
@@ -236,3 +246,41 @@ void IsisMain() {
...
+  inCube.open(file, "r");
+
+  LineManager line(inCube);
+  Statistics cubeStats;
+
+  for (int i = 1; i <= inCube.lineCount(); i++) {
+    line.SetLine(i);
+    inCube.read(line);
+    cubeStats.AddData(line.DoubleBuffer(), line.size());
+  }
```
Comment (jessemapel) [inline isis/src/base/apps/hist/main.cpp:275]: Is there a reason why you manually did this instead of using the Cube::statistics function?

**Review by jessemapel — COMMENTED** [review]
What is the expected behavior for multi-band cubes? This just checks the first band.

**Review by AustinSanders — APPROVED** [review]

File: isis/src/base/apps/hist/main.cpp (hunk 1 above)
Comment (scsides) [reply isis/src/base/apps/hist/main.cpp:275]: The cube stats and hist don't work inside the GuiHelper environment. Not sure why, thought I'd investigate later.

Comment (jessemapel) [reply isis/src/base/apps/hist/main.cpp:275]: Strange and kind of concerning, but not an issue for this PR.

**Comment by scsides** [comment]
@jessemapel - The hist program only works on one band. If more exist or are specified in an input attribute, the first is used and all others are ignored. I'm not sure what the use case would be for a single histogram with data from multiple bands would be. It could output multiple hists (one for each band), but that's not the way it was written and it would have to allow a different min/max/nbins for each band (a hyperspectral cube could be difficult). The Gui Helper follows the way the current app works.

**Pull request merged**